### PR TITLE
[SMALLFIX] creating LockResource

### PR DESCRIPTION
**Pull request by jsimsa**
This PR create LockResource, which makes it possible to acquire and release locks using the try-with-resources idiom.

### CODE REVIEW

File: core/common/src/test/java/alluxio/resource/LockResourceTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        lock.unlock();
+      }
+    }
+  }
+
+  /**
+   * Tests {@link LockResource} with {@link ReentrantReadWriteLock}.
+   */
+  @Test
+  public void reentractReadWriteLockTest() {
```
Comment (gpang) [inline core/common/src/test/java/alluxio/resource/LockResourceTest.java]: reentract -> reentrant

**Comment by gpang** [comment]
I left a minor comment, but looks good. Thanks!

File: core/common/src/test/java/alluxio/resource/LockResourceTest.java (hunk 1 above)
Comment (jsimsa) [inline core/common/src/test/java/alluxio/resource/LockResourceTest.java]: done

**Comment by gpang** [comment]
LGTM

**Pull request merged**